frontend/src/components/elements/PlotlyChart/CustomTheme.tsx

```diff
@@ -0,0 +1,640 @@
+ * limitations under the License.
+ */
+
+import { merge, assign } from "lodash"
+
+import { useTheme } from "@emotion/react"
+
+import { hasLightBackgroundColor, Theme } from "src/theme"
+
+// TODO: for these colors below, these likely need to move to our theme!
+// For the meantime, these colors will be defined for plotly.
+
+const divergingColorscaleLightTheme = [
+  [0.1, "#004280"],
+  [0.2, "#0054A3"],
+  [0.3, "#1C83E1"],
+  [0.4, "#60B4FF"],
+  [0.5, "#A6DCFF"],
+  [0.6, "#FFC7C7"],
+  [0.7, "#FF8C8C"],
+  [0.8, "#FF4B4B"],
+  [0.9, "#BD4043"],
+  [1.0, "#7D353B"],
+]
+
+const divergingColorscaleDarkTheme = [
+  [0, "#A6DCFF"],
+  [0.1, "#A6DCFF"],
+  [0.2, "#60B4FF"],
+  [0.3, "#1C83E1"],
+  [0.4, "#0054A3"],
+  [0.5, "#004280"],
+  [0.6, "#7D353B"],
+  [0.7, "#BD4043"],
+  [0.8, "#FF4B4B"],
+  [0.9, "#FF8C8C"],
+  [1.0, "#FFC7C7"],
+]
+
+const sequentialColorscaleLightTheme = [
+  [0, "#E4F5FF"],
+  [0.1111111111111111, "#C7EBFF"],
+  [0.2222222222222222, "#A6DCFF"],
+  [0.3333333333333333, "#83C9FF"],
+  [0.4444444444444444, "#60B4FF"],
+  [0.5555555555555556, "#3D9DF3"],
+  [0.6666666666666666, "#1C83E1"],
+  [0.7777777777777778, "#0068C9"],
+  [0.8888888888888888, "#0054A3"],
+  [1, "#004280"],
+]
+
+const sequentialColorscaleDarkTheme = [
+  [0, "#004280"],
+  [0.1111111111111111, "#0054A3"],
+  [0.2222222222222222, "#0068C9"],
+  [0.3333333333333333, "#1C83E1"],
+  [0.4444444444444444, "#3D9DF3"],
+  [0.5555555555555556, "#60B4FF"],
+  [0.6666666666666666, "#83C9FF"],
+  [0.7777777777777778, "#A6DCFF"],
+  [0.8888888888888888, "#C7EBFF"],
+  [1, "#E4F5FF"],
+]
+
+const categoryColorsLightTheme = [
+  "#0068C9",
+  "#83C9FF",
+  "#FF2B2B",
+  "#FFABAB",
+  "#29B09D",
+  "#7DEFA1",
+  "#FF8700",
+  "#FFD16A",
+  "#6D3FC0",
+  "#D5DAE5",
+]
+
+const categoryColorsDarkTheme = [
+  "#83C9FF",
+  "#0068C9",
+  "#FFABAB",
+  "#FF2B2B",
+  "#7DEFA1",
+  "#29B09D",
+  "#FFD16A",
+  "#FF8700",
+  "#6D3FC0",
+  "#D5DAE5",
+]
+
+export function getGray70(theme: Theme): string {
+  return hasLightBackgroundColor(theme)
+    ? theme.colors.gray70
+    : theme.colors.gray30
+}
+
+export function getGray30(theme: Theme): string {
+  return hasLightBackgroundColor(theme)
+    ? theme.colors.gray30
+    : theme.colors.gray85
+}
+
+export function getGray90(theme: Theme): string {
+  return hasLightBackgroundColor(theme)
+    ? theme.colors.gray90
+    : theme.colors.gray10
+}
+
+export function getDecreasingRed(theme: Theme): string {
+  return hasLightBackgroundColor(theme) ? "#FF2B2B" : "#FFABAB"
+}
+
+export function getIncreasingGreen(theme: Theme): string {
+  return hasLightBackgroundColor(theme) ? "#29B09D" : "#7DEFA1"
+}
+
+/**
+ * This applies categorical colors (discrete or labeled data) to
+ * graphs by mapping legend groups to marker colors and customdata to marker colors.
+ * This is done because colorway is not fully respected by plotly.
+ * @param data - spec.data
+ */
+export function changeDiscreteColors(data: any): void {
```

Comment:
:(

Reply:
Is it possible if we pin the version of plotly.js, this could make it so that we don't run into a bunch of problems later on and manually update plotly.js?

Reply:
the problem is mostly regarding the spec coming from the backend and I don't think we should pin it on the backend